samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/DocumentImportController.cs

```diff
@@ -76,108 +76,182 @@ private enum SupportedFileType
+
+        if (formFiles.Count == 0)
+        {
+            throw new ArgumentException("No file was uploaded.");
+        }
+        else if (formFiles.Count > this._options.FileCountLimit)
+        {
+            throw new ArgumentException($"Too many files uploaded. Max file count is {this._options.FileCountLimit}.");
+        }
+
+        // Loop through the uploaded files and validate them before importing.
+        foreach (var formFile in formFiles)
+        {
+            if (formFile.Length == 0)
+            {
+                throw new ArgumentException($"File {formFile.FileName} is empty.");
+            }
+
+            if (formFile.Length > this._options.FileSizeLimit)
+            {
+                throw new ArgumentException($"File {formFile.FileName} size exceeds the limit.");
+            }
+
+            // Make sure the file type is supported.
+            var fileType = this.GetFileType(Path.GetFileName(formFile.FileName));
+            switch (fileType)
+            {
+                case SupportedFileType.Txt:
+                case SupportedFileType.Pdf:
+                    break;
+                default:
+                    throw new ArgumentException($"Unsupported file type: {fileType}");
+            }
+        }
+
+        // Make sure the user has access to the chat session if the document is uploaded to a chat session.
         if (documentImportForm.DocumentScope == DocumentImportForm.DocumentScopes.Chat
-            && !(await this.UserHasAccessToChatAsync(documentImportForm.UserId, documentImportForm.ChatId)))
+                && !(await this.UserHasAccessToChatAsync(documentImportForm.UserId, documentImportForm.ChatId)))
         {
-            return this.BadRequest("User does not have access to the chat session.");
+            throw new ArgumentException("User does not have access to the chat session.");
         }
+    }
 
+    /// <summary>
+    /// Import a single document.
+    /// </summary>
+    /// <param name="kernel">The kernel.</param>
+    /// <param name="formFile">The form file.</param>
+    /// <param name="documentImportForm">The document import form.</param>
+    /// <returns>True if the file is imported successfully. False otherwise.</returns>
+    private async Task<bool> ImportDocumentHelperAsync(IKernel kernel, IFormFile formFile, DocumentImportForm documentImportForm)
+    {
         var fileType = this.GetFileType(Path.GetFileName(formFile.FileName));
-        var fileContent = string.Empty;
+        var documentContent = string.Empty;
         switch (fileType)
         {
             case SupportedFileType.Txt:
-                fileContent = await this.ReadTxtFileAsync(formFile);
+                documentContent = await this.ReadTxtFileAsync(formFile);
                 break;
             case SupportedFileType.Pdf:
-                fileContent = this.ReadPdfFile(formFile);
+                documentContent = this.ReadPdfFile(formFile);
```

Comment:
should this also be an async function ?

Reply:
This is not because the PdfPig doesn't have an async API.